static-analysis-plugin/plugin/src/main/groovy/com/novoda/staticanalysis/CheckstyleConfigurator.groovy

```diff
@@ -0,0 +1,68 @@
+package com.novoda.staticanalysis
+
+import groovy.util.slurpersupport.GPathResult
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import org.gradle.api.plugins.quality.Checkstyle
+import org.gradle.internal.logging.ConsoleRenderer
+
+class CheckstyleConfigurator {
+
+    void configure(Project project, StaticAnalysisExtension extension) {
+        project.apply plugin: 'checkstyle'
+        project.afterEvaluate {
+            PenaltyExtension penalty = extension.penalty
+            boolean isAndroidApp = project.plugins.hasPlugin('com.android.application')
+            boolean isAndroidLib = project.plugins.hasPlugin('com.android.library')
+            if (isAndroidApp || isAndroidLib) {
+                def variants = isAndroidApp ? project.android.applicationVariants : project.android.libraryVariants
+                configureAndroid(project, variants)
+            }
+            Map<String, Integer> violations = [errors: 0, warnings: 0]
+            List<String> reports = []
+            project.tasks['check'].doLast {
```

Comment:
Well, I wanted to avoid to create another task for this (as in: I did it, then removed it in favor of this approach). For instance adding a task to evaluate the reports means defining a `dependsOn` relation against every `Checkstyle` task too. I thought this _shortcut_  would be less intrusive 💃